NameError: global name 'actions_yaml_file' is not defined

The latest charm tools is broken, and has in turn broken the gate for OpenStack charm testing and code landing upstream.  Please consider an urgent revert or fix.  Thank you.


```
INFO:root:Workspace dir: /var/lib/jenkins/workspace/test_charm_lint
INFO:root:Reading file: Makefile
INFO:root:Searching for: ['@flake8', '/flake8', 'tox -e pep8']
INFO:root:command:  make -f Makefile lint
pep8 create: /var/lib/jenkins/checkout/cinder/.tox/pep8
pep8 installdeps: -r/var/lib/jenkins/checkout/cinder/requirements.txt, -r/var/lib/jenkins/checkout/cinder/test-requirements.txt
pep8 runtests: commands[0] | flake8 hooks unit_tests tests actions
pep8 runtests: commands[1] | charm-proof
Traceback (most recent call last):
  File ".tox/pep8/bin/charm-proof", line 9, in <module>
    load_entry_point('charm-tools==2.1.0', 'console_scripts', 'charm-proof')()
  File "/var/lib/jenkins/checkout/cinder/.tox/pep8/local/lib/python2.7/site-packages/charmtools/proof.py", line 62, in main
    lint, exit_code = proof(args_.charm_name, args_.bundle, args_.debug)
  File "/var/lib/jenkins/checkout/cinder/.tox/pep8/local/lib/python2.7/site-packages/charmtools/proof.py", line 56, in proof
    lint, err_code = c.proof()
  File "/var/lib/jenkins/checkout/cinder/.tox/pep8/local/lib/python2.7/site-packages/charmtools/charms.py", line 416, in proof
    if os.path.exists(actions_yaml_file):
NameError: global name 'actions_yaml_file' is not defined
ERROR: InvocationError: '/var/lib/jenkins/checkout/cinder/.tox/pep8/bin/charm-proof'
___________________________________ summary ____________________________________
ERROR:   pep8: commands failed
make: *** [lint] Error 1
ERROR:root:Make target returned non-zero.
```

--- a/charmtools/charms.py
+++ b/charmtools/charms.py
@@ -659,7 +659,7 @@ def validate_actions(actions, action_hooks, linter):
 
     # TODO: Schema validation
     for k in actions:
-        if k.startswith('juju-'):
+        if k.startswith('juju'):
             linter.err('actions.{}: juju is a reserved namespace'.format(k))
             continue
         h = os.path.join(action_hooks, k)

--- a/charmtools/charms.py
+++ b/charmtools/charms.py
@@ -259,7 +259,7 @@ class Charm(object):
         hooks_path = os.path.join(charm_path, 'hooks')
         actions_path = os.path.join(charm_path, 'actions')
         yaml_path = os.path.join(charm_path, 'metadata.yaml')
-        actions_yaml_path = os.path.join(charm_path, 'actions.yaml')
+        actions_yaml_file = os.path.join(charm_path, 'actions.yaml')
         try:
             yamlfile = open(yaml_path, 'r')
             try:
@@ -418,7 +418,7 @@ class Charm(object):
                     try:
                         actions = yaml.safe_load(f.read())
                     except Exception as e:
-                        lint.crit('cannot parse ' + actions_yaml_path + ":" + str(e))
+                        lint.crit('cannot parse ' + actions_yaml_file + ":" + str(e))
                     validate_actions(actions, actions_path, lint)
 
         except IOError:

--- a/charmtools/charms.py
+++ b/charmtools/charms.py
@@ -653,16 +653,12 @@ def validate_actions(actions, action_hooks, linter):
     if not actions:
         return
 
-    if 'actions' not in actions:
-        linter.err("actions: actions.yaml needs to declare an 'actions:' key")
-        return
-
-    if not isinstance(actions['actions'], dict):
+    if not isinstance(actions, dict):
         linter.err('actions: must be a dictionary of json schemas')
         return
 
     # TODO: Schema validation
-    for k in actions['actions']:
+    for k in actions:
         if k.startswith('juju-'):
             linter.err('actions.{}: juju is a reserved namespace'.format(k))
             continue

--- a/charmtools/charms.py
+++ b/charmtools/charms.py
@@ -665,7 +665,7 @@ def validate_actions(actions, action_hooks, linter):
         h = os.path.join(action_hooks, k)
         if not os.path.isfile(h):
             linter.warn('actions.{0}: actions/{0} does not exist'.format(k))
-        elif not os.access(fpath, os.X_OK):
+        elif not os.access(h, os.X_OK):
             linter.err('actions.{0}: actions/{0} is not executable'.format(k))
 
 

--- a/tests/test_charm_proof.py
+++ b/tests/test_charm_proof.py
@@ -668,7 +668,7 @@ class ActionsValidationTest(TestCase):
         """Charm has the minimum allowed actions configuration."""
         linter = Mock()
         actions = {
-            'actions': {}
+            'an-action': {}
         }
         validate_actions(actions, 'actions', linter)
         self.assertFalse(linter.err.called)
@@ -677,14 +677,12 @@ class ActionsValidationTest(TestCase):
         """Charm has multiple actions."""
         linter = Mock()
         actions = {
-            'actions': {
-                'do': {
-                    'description': 'a thing',
-                },
-                'do-not': {
-                    'description': 'not a thing',
-                },
-            }
+            'do': {
+                'description': 'a thing',
+            },
+            'do-not': {
+                'description': 'not a thing',
+            },
         }
         with patch('os.path.exists'):
             validate_actions(actions, 'actions', linter)
@@ -694,14 +692,12 @@ class ActionsValidationTest(TestCase):
         """Charm has multiple actions."""
         linter = Mock()
         actions = {
-            'actions': {
-                'juju-do': {
-                    'description': 'a thing',
-                },
-                'do-not': {
-                    'description': 'not a thing',
-                },
-            }
+            'juju-do': {
+                'description': 'a thing',
+            },
+            'do-not': {
+                'description': 'not a thing',
+            },
         }
 
         with patch('os.path.exists'):

